velox/exec/Spill.cpp

```diff
@@ -327,17 +336,41 @@ tsan_atomic<uint32_t>& injectedSpillCount() {
   return injectedCount;
 }
 
-bool testingTriggerSpill() {
+bool testingTriggerSpill(const std::string& name) {
   // Do not evaluate further if trigger is not set.
   if (testingSpillPct() <= 0) {
     return false;
   }
-  if (folly::Random::rand32() % 100 > testingSpillPct()) {
+
+  if (injectedSpillCount() >= maxSpillInjections()) {
     return false;
   }
-  if (injectedSpillCount() >= maxSpillInjections()) {
+
+  if (!name.empty() && !testingSpillOperators().empty()) {
+    std::unordered_set<std::string> tokens;
+    std::istringstream iss(name);
+    std::string token;
+    while (std::getline(iss, token, '.')) {
+      tokens.insert(token);
+    }
+
+    bool mayNotSpill = true;
```

Comment:
Let's check pool first? And we pass regular expression instead of list of pool names here? Thanks!